components/src/md3/Header/Header.tsx

```diff
@@ -0,0 +1,748 @@
+import React, { useCallback, useState, useRef, useEffect, ReactNode } from 'react';
+import {
+    Animated,
+    ImageSourcePropType,
+    SafeAreaView,
+    StyleSheet,
+    StatusBar,
+    TextInput,
+    TouchableWithoutFeedback,
+    ViewProps,
+    StyleProp,
+    ViewStyle,
+    TextStyle,
+    ImageStyle,
+    TextInputProps,
+    Platform,
+} from 'react-native';
+import color from 'color';
+import { MD3Theme, useTheme } from 'react-native-paper';
+import { ANIMATION_LENGTH } from './constants';
+import { HeaderBackgroundImage } from './headerBackgroundImage';
+import { HeaderNavigationIcon } from './headerNavigationIcon';
+import { HeaderContent } from './headerContent';
+import { HeaderActionItems } from './headerActionItems';
+import { SearchContext } from './contexts/SearchContextProvider';
+import { ColorContext } from './contexts/ColorContextProvider';
+import { HeaderHeightContext } from './contexts/HeaderHeightContextProvider';
+import { HeaderActionComponent, HeaderIcon, IconSource } from '../__types__';
+import { $DeepPartial } from '@callstack/react-theme-provider';
+import { usePrevious } from '../__hooks__/usePrevious';
+import { useHeaderDimensions } from '../__hooks__/useHeaderDimensions';
+import { useFontScale } from '../__contexts__/font-scale-context';
+const AnimatedSafeAreaView = Animated.createAnimatedComponent(SafeAreaView);
+
+const headerStyles = (
+    props: HeaderProps,
+    theme: MD3Theme,
+    fontScale: number
+): StyleSheet.NamedStyles<{
+    root: ViewStyle;
+    content: ViewStyle;
+    search: ViewStyle;
+}> => ({
+    root: {
+        width: '100%',
+        backgroundColor:
+            props.backgroundColor ||
+            // @change color once have a correct color
+            // (theme.dark ? theme.colors.actionPalette?.active || theme.colors.surface : theme.colors.primary),
+            (theme.dark ? theme.colors.surface : theme.colors.primary),
+        shadowColor: 'rgba(0, 0, 0, 0.3)',
+        shadowOffset: {
+            width: 0,
+            height: 1,
+        },
+        shadowRadius: 2,
+        shadowOpacity: 1,
```

Comment:
Is there any chance that we can include shadow styles in a theme? If so, can you write a story/modify an existing story to do it?

Reply:
I think that's not possible to include shadow in theme as no properties are present
the theme consists on mainly FONT and COLOUR